beacon-chain/sync/initial-sync/sync_blocks.go

```diff
@@ -0,0 +1,172 @@
+package initialsync
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+)
+
+func (s *InitialSync) processBlockAnnounce(msg p2p.Message) {
+	_, span := trace.StartSpan(msg.Ctx, "beacon-chain.sync.initial-sync.processBlockAnnounce")
+	defer span.End()
+	data := msg.Data.(*pb.BeaconBlockAnnounce)
+	recBlockAnnounce.Inc()
+
+	if s.stateReceived && data.SlotNumber > s.highestObservedSlot {
+		s.requestBatchedBlocks(s.lastRequestedSlot, data.SlotNumber)
+		s.lastRequestedSlot = data.SlotNumber
+	}
+}
+
+// processBlock is the main method that validates each block which is received
+// for initial sync. It checks if the blocks are valid and then will continue to
+// process and save it into the db.
+func (s *InitialSync) processBlock(ctx context.Context, block *pb.BeaconBlock) {
+	ctx, span := trace.StartSpan(ctx, "beacon-chain.sync.initial-sync.processBlock")
+	defer span.End()
+	recBlock.Inc()
+
+	if block.Slot == s.highestObservedSlot {
+		s.currentSlot = s.highestObservedSlot
+		if err := s.exitInitialSync(s.ctx); err != nil {
+			log.Errorf("Could not exit initial sync: %v", err)
+			return
+		}
+		return
+	}
+
+	if block.Slot < s.currentSlot {
```

Comment:
If we have a block older than the current slot counter we use for tracking initial sync, we disregard